src/worker/worker/collectors/rss_collector.py

```diff
@@ -83,20 +106,26 @@ def parse_feed_entry(self, feed_entry: feedparser.FeedParserDict, source) -> New
         title: str = str(feed_entry.get("title", ""))
         description: str = str(feed_entry.get("description", ""))
         link: str = str(feed_entry.get("link", ""))
+
         if link_transformer := source["parameters"].get("LINK_TRANSFORMER", None):
             link = self.link_transformer(link, link_transformer)
 
         published = self.get_published_date(feed_entry)
-
-        content_location = source["parameters"].get("CONTENT_LOCATION", None)
-        content_from_feed, content_location = self.content_from_feed(feed_entry, content_location)
-        content = str(feed_entry[content_location]) if content_from_feed else ""
-        if link:
-            web_content = self.extract_web_content(link, self.xpath)
-            content = content if content_from_feed else str(web_content.get("content"))
-            author = author or str(web_content.get("author"))
-            title = title or str(web_content.get("title"))
-            published = published or web_content.get("published_date") or self.last_modified
+        content = ""
+
+        if self.use_feed_content:
+            content = self.extract_content_from_feed(feed_entry)
+
+            if self.xpath and content:
+                if extracted := self.xpath_extraction(content, self.xpath):
+                    content = extracted
+        else:
+            if link:
+                web_content = self.extract_web_content(link, self.xpath)
+                content = str(web_content.get("content"))
+                author = author or str(web_content.get("author"))
+                title = title or str(web_content.get("title"))
+                published = published or web_content.get("published_date") or self.last_modified
```

Comment:
To cover an edge case where nothing can be extracted

```suggestion
        content = ""

        if self.use_feed_content:
            content = self.extract_content_from_feed(feed_entry)

            if self.xpath and content:
                if extracted := self.xpath_extraction(content, self.xpath):
                    content = extracted
        elif link:
                web_content = self.extract_web_content(link, self.xpath)
                content = str(web_content.get("content"))
                author = author or str(web_content.get("author"))
                title = title or str(web_content.get("title"))
                published = published or web_content.get("published_date") or self.last_modified
        else:
             logger.warning("No content could be extracted")
```